### PR TITLE
New option 'autoGrayBaseLayer' 

### PR DESCRIPTION
**Pull request by eddy-geek**
...for better readability when using slope overlays on top of colorful maps
> When checked and an overlay is enabled, base layer is displayed in gray

image

### CODE REVIEW

File: js/wtracks.js [hunk 1]
```diff
@@ -2147,6 +2153,26 @@ $(function(){
...
+    console.log("wt.grayscaleBaseLayer", grayscaleBaseLayer)
+    storeVal("wt.grayscaleBaseLayer", grayscaleBaseLayer);
+    setGrayscaleBaseLayer();
+  });
+
+  function setGrayscaleBaseLayer() {
+    objectForEach(map._layers, function(lId, layer) {
+      if (layer._container) {
+        if (grayscaleBaseLayer) {
+          if (! layer._container.classList.contains('blend-multiply'))  // not an overlay
```
Comment (eddy-geek) [inline js/wtracks.js:2167]: not very happy with this. do you see a cleaner, simple way to detect overlays?

Comment (eddy-geek) [reply js/wtracks.js:2167]: reworked.

**Comment by opoto** [comment]
I see you're using internal leaflet API (_layers, _container). This should only be done when there is no alternative with public API. Did you check?

**Comment by eddy-geek** [comment]
Good point. I replaced `_container` by `getContainer()` ; but for `_layer` it's harder.

You had the same issue this month for [the opacity slider](https://github.com/opoto/wtracks/commit/74c650b5). See also Leaflet/Leaflet#3126.

**Comment by opoto** [comment]
Thanks!

**Pull request merged**